plugins/out_es/es.c

```diff
@@ -311,6 +313,10 @@ static int elasticsearch_format(struct flb_config *config,
     msgpack_unpacked_destroy(&result);
     msgpack_unpacked_init(&result);
 
+    flb_sds_t write_op = strcasecmp(ctx->write_operation, FLB_ES_WRITE_OP_UPSERT) == 0
```

Comment:
Could you check it on init ? How about adding write_op to `flb_elasticsearch` ?
It will take cost if we do on runtime.